docs/source/widgets/index.rst

```diff
@@ -0,0 +1,105 @@
+###################
+Widgets
+###################
+
+The AiiDAlab largely dependents on the Jupyter widgets for the graphical user interface (GUI), namely the `Ipywidgets <https://ipywidgets.readthedocs.io/>`_ package. 
+Furthremore, we create the `aiidalab-widgets-base <https://github.com/aiidalab/aiidalab-widgets-base>`_ to facilitate to develop apps in AiiDAlab.
+
+***********************
+Ipywidgets
+***********************
+
+`Ipywidgets <https://ipywidgets.readthedocs.io/>`_ is a python package to create widgets inside the Jupyter. 
+Widgets are eventful python objects, which shows GUI components such as a slider, button, etc.
+
+Ipywidgets provides base widgets like sliders, progressbar, buttons, checkbox, dropdown, radio buttons, text, textarea, label, etc. 
+The widgets have properties and value, which can be connected to the python codes and upated dynamically.
+For a simple example, one can use a slider to control the value of a variable.
+
+.. code-block:: python
+
+    from ipywidgets import FloatSlider
+
+    w = FloatSlider(value=0.0, min=0.0, max=10.0, description="b:");
+    a = 0.0;
+
+    def slider_change(change):
+        global a
+
+        print(change["old"])
+        print(change["new"])
+
+        a = a + w.value
+
+    a.obersve(slider_change, names="value")
+
+In this example, the ``slider_change`` function will be called each time the slider' value being updated.
+
+Traitlets
+=================
+
+In ipywidgets, all the widgets are using the `traitlets <https://traitlets.readthedocs.io/>`_ types for the parameters.
+As shown in the example above, the value of the FloatSlider is a Float trait type.
+One can not only show the current value ``value["new"]`` but also the old value ``value["old"]``.
+Traits can emit change events when the attributes are upated. This is very useful for creating custom widgets.
+
+Custom widgets
+=================
+
+Ipywidgets package also supports for developers to create their own widgets.
+There is a detailed `tutorial <https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Custom.html>`_ to show how to develop the custom widgets.
+In the ipywidgets framework is based on the Comm framework.
```

Comment:
Please add a link.

Reply:
The tutorial is hyperlink for the tutorial.